### PR TITLE
BI-24: Write a CI/CD pipeline for `web-app-client`

### PR DESCRIPTION
**Pull request by ayushtiwari110**
Added github workflows for:
- running prettier, lint and checking build
- deploying storybook to chromatic

### CODE REVIEW

**Comment by ayushtiwari110** [comment]
The Deploy StoryBook is failing currently, because secret variables are not yet there in project settings. Once they are there, I guess it must be fine...

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: npm run build
+
+      - name: Check Build Status
+        run: |
+          if [ $? -eq 0 ]; then
+            echo "Build successful!"
+          else
+            echo "Build failed!"
+            exit 1
+          fi
```
Comment (bacpactech) [inline .github/workflows/build.yml:37]: Let's try to combine these two `.yml` files into one. It'll allow us to define an order in which both of them will execute, and we'll use the same code to trigger the script.

To do that all you need to do is to migrate the code from line number 9 to 30 of `chromatic.yml` to this file. Make sure the indentation is such that:

```yml
jobs:
   build-project:
      name: Run Prettier, Lint and Build Check
      // your first amazing automation script
   chromatic-deployment:
      name: Deploy StoryBook on Chromatic
      // your second amazing automation script
```

You can use the `name:` attribute to give name to the two jobs.  Once you do that, please rename this file to `actions.yml`.

You can use the `git mv` command to do that.

**Review by bacpactech — CHANGES_REQUESTED** [review]

File: .github/workflows/chromatic.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Install dependencies
+        run: npm install
+        # 👇 Adds Chromatic as a step in the workflow
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
+        # Chromatic GitHub Action options
+        with:
+          # 👇 Chromatic projectToken, refer to the manage page to obtain it.
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (ayushtiwari110) [reply .github/workflows/chromatic.yml:30]: Thanks!

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Runs Preetier Check, Lint Check, Build Check
```
[the remark this answers is not in the document]
Comment (ayushtiwari110) [reply .github/workflows/build.yml:1]: yeah sure! I have renamed it.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# first runs prettier and lint checks then builds the code.
+on: [pull_request, push]
```
[the remark this answers is not in the document]
Comment (ayushtiwari110) [reply .github/workflows/build.yml:3]: done!

File: .github/workflows/build.yml (hunk 1 above)
Comment (ayushtiwari110) [reply .github/workflows/build.yml:37]: ok sure!

File: package.json [hunk 5]
```diff
@@ -9,7 +9,8 @@
     "lint": "next lint",
     "storybook": "storybook dev -p 6006",
     "build-storybook": "storybook build",
-    "chromatic": "npx chromatic --project-token=chpt_1a0522d85a68f99"
+    "chromatic": "npx chromatic --project-token=chpt_1a0522d85a68f99",
```
[the remark this answers is not in the document]
Comment (ayushtiwari110) [reply package.json:12]: Yeah we don't need it. Actually i also thought the same after making the PR. Removed it now

File: .github/workflows/actions.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        run: |
+          if [ $? -eq 0 ]; then
+            echo "Build successful!🎉"
+          else
+            echo "Build failed!"
+            exit 1
+          fi
+
+  chromatic-deployment: #Second Job is to deploy the project to Chromatic
+    # Operating System
```
Comment (bacpactech) [inline .github/workflows/actions.yml:42]: Please add the `name` attribute over here.

**Review by bacpactech — CHANGES_REQUESTED** [review]

File: .github/workflows/actions.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
+name: Actions Workflow
```
Comment (bacpactech) [inline .github/workflows/actions.yml:1]: Now, come to think of it, this label is not helping us at all. We can remove this line.

**Review by bacpactech — CHANGES_REQUESTED** [review]

File: .github/workflows/actions.yml (hunk 7 above)
Comment (ayushtiwari110) [reply .github/workflows/actions.yml:1]: ok removed

File: .github/workflows/actions.yml (hunk 6 above)
Comment (ayushtiwari110) [reply .github/workflows/actions.yml:42]: done!

**Review by bacpactech — APPROVED** [review]
Good Job! 🎉

**Pull request merged**